pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,409 @@ import (
+			resultBuffer := make([]byte, len(want))
+			By("Reading data from buffer")
+			ds.Read(resultBuffer)
+			Expect(ds.accessKeyID).To(Equal(accessKeyID))
+			Expect(ds.secretKey).To(Equal(secretKey))
+			Expect(ds.qemu).To(Equal(qemu))
+			Expect(reflect.DeepEqual(resultBuffer, want)).To(BeTrue())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		table.Entry("expect NewDataStream to succeed with valid image", "cirros-qcow2.img", "", "", true, cirrosData, false),
+		table.Entry("expect NewDataStream to fail with non existing image", "badimage.iso", "", "", false, nil, true),
+		table.Entry("expect NewDataStream to fail with invalid or missing image", "", "", "", false, nil, true),
+		table.Entry("expect NewDataStream to succeed with valid iso image", "tinyCore.iso", "accessKey", "secretKey", false, tinyCoreData, false),
+	)
+
+	It("can close all readers", func() {
+		ds, err := NewDataStream(ts.URL+"/tinyCore.iso", "", "")
+		Expect(err).NotTo(HaveOccurred())
+		emptyBuffer(ds)
+		By("Closing data stream")
+		err = ds.Close()
+		Expect(err).NotTo(HaveOccurred())
+	})
 
-func NewQEMUAllErrors() image.QEMUOperations {
-	err := errors.New("qemu should not be called from this test override with replaceQEMUOperations")
-	return NewFakeQEMUOperations(err, err, err)
-}
+	table.DescribeTable("can build selector", func(image, ep string, wantErr bool) {
+		if ep == "" {
+			ep = ts.URL
+		}
+		ds, err := NewDataStream(ep+"/"+image, "", "")
+		if !wantErr {
+			Expect(err).NotTo(HaveOccurred())
+			defer ds.Close()
+			emptyBuffer(ds)
+			err = ds.dataStreamSelector()
+			Expect(err).NotTo(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		table.Entry("success building selector for valid endpoint and file", "tinyCore.iso", "", false),
+		table.Entry("fail trying to build selector for invalid http endpoint", "", "http://www.google.com", true),
+		table.Entry("fail trying to build invalid selector", "tinyCore.iso", "fake://somefakefile", true),
+	)
+
+	table.DescribeTable("can construct readers", func(outfile, filename string, numRdrs int, wantErr bool) {
+		fmt.Println(filename)
+		By(fmt.Sprintf("Creating new fileserver for %s and file %s", filepath.Dir(filename), filepath.Base(filename)))
+		tempTestServer := createTestServer(filepath.Dir(filename))
+		By(fmt.Sprintf("Creating new datastream to %s", tempTestServer.URL+"/"+filepath.Base(filename)))
+		ds, err := NewDataStream(tempTestServer.URL+"/"+filepath.Base(filename), "", "")
+		defer func() {
+			tempTestServer.Close()
+		}()
+
+		if !wantErr {
+			Expect(err).NotTo(HaveOccurred())
+			defer ds.Close()
+			for _, r := range ds.Readers {
+				fmt.Fprintf(GinkgoWriter, "INFO: Reader type: %d\n", r.rdrType)
+			}
+			Expect(len(ds.Readers)).To(Equal(numRdrs))
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		table.Entry("successfully construct a xz reader", "tinyCore.iso.xz", testfiles[".xz"], 5, false),              // [http, multi-r, xz, multi-r]
```

Comment:
+1, nice approach here!

Reply:
Thanks the original tests were all in table form already, I just converted them to the DescribeTable versions of gingko